### PR TITLE
Add flake8-numbers plugin to formatting step

### PR DESCRIPTION
**Pull request by Hespe**
## Description

This change adds the flake8-numbers plugin to ensure all number literals are properly formatted.
The `tests/resources` and `docs/examples` directories are excluded, since they contain lattice files that should not be linted.

## Motivation and Context

- [x] I have raised an issue to propose this change (#184)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (_required for a bug fix or a new feature_).
- [ ] I have updated the documentation accordingly.
  
  
- [x] I have reformatted the code and checked that formatting passes (**required**).
- [x] I have have fixed all issues found by `flake8` (**required**).
- [x] I have ensured that all `pytest` tests pass (**required**).
- [ ] I have run `pytest` on a machine with a CUDA GPU and made sure all tests pass (**required**).
- [ ] I have checked that the documentation builds (**required**).

Note: We are using a maximum length of 88 characters per line

### CODE REVIEW

File: .github/workflows/format.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.10"
+          python-version: "3.11"
```
Comment (jank324) [inline .github/workflows/format.yaml:14]:
```suggestion
          python-version: "3.12"
```

**Comment by jank324** [comment]
It seems a little like `flake8-numbers` doesn't work all that well yet.

Technically, Python doesn't care where you put digit separators. It seems `flake8-numbers` does and it wants you to do e.g. `3.14_159_265_359` But I feel like `3.141_592_653_59` is more intuitive(?) The latter seems to also be what GitHub Copilot wants to do.

**Comment by Hespe** [comment]
I agree, I would have preferred to start counting the digits from the decimal separator, to the front and to the back. Starting from the last digit seems wrong.

**Comment by jank324** [comment]
I converted this back to draft for now, because with the current state of the `flake8-numbers` plugin, I think we should defer on using it.

**Comment by jank324** [comment]
Closing this because the plug-in is not in a state for us to use it.

**Pull request closed without merging**